File: 06/objc/driver.m
#import "vehicle.h"
#import "taxi.h"
#import "truck.h"
#import "parkinglot.h"

int main(int argc, char* argv[]) {
	Vehicle *car = [[Vehicle alloc] AmountOfDoor: 2 AmountOfCylinder: 6 TransmissionType: 3 Color: @"blue" FuelLevel: 14.6];
	Taxi *taxi = [[Taxi alloc] AmountOfDoor: 4 AmountOfCylinder: 6 TransmissionType: 5 Color: @"yellow" FuelLevel: 3.3 PresenceOfCustomer: NO];
	Truck *truck = [[Truck alloc] AmountOfDoor: 2 AmountOfCylinder: 16 TransmissionType: 8 Color: @"black" FuelLevel: 7.54 IfLoaded: YES];
	ParkingLot *lot = [ParkingLot new];
	printf("Message passing:\n");
	[car printDetail];
	printf("\n");
	[taxi printDetail];
	printf("\n");
	[truck printDetail];
	printf("Horn!\n");
	[car horn];
	[taxi horn];
	[truck horn];
	[lot addVehicle: car];
	[lot addVehicle: taxi];
	printf("Parking Lot Details:\n");
	[lot printVehicles];
	printf("Horn each vehicle in the parking lot:\n");
	[lot hornEach];
	return 0;
}

File: 06/driver.cpp
#include "vehicle.h"
#include "taxi.h"
#include "truck.h"
#include "parkinglot.h"

using namespace std;

int main(int argc, char* argv[]) {
	Vehicle *car = new Vehicle(2, 6, 3, "blue", 14.6);
	Vehicle *taxi = new Taxi(4, 6, 5, "yellow", 3.3, false);
	Vehicle *truck = new Truck(2, 16, 8, "black", 7.54, true);
	ParkingLot *lot = new ParkingLot();
	cout << "Regular Method Calling:" << endl;
	car->printDetail();
	cout << endl;
	taxi->printDetail();
	cout << endl;
	truck->printDetail();
	cout << endl;
	cout << "Operator Overloading:" << endl;
	cout << *car << endl;
	cout << *taxi << endl;
	cout << *truck << endl;
	cout << "Horn!" << endl;
	car->horn();
	taxi->horn();
	truck->horn();
	lot->addVehicle(car);
	lot->addVehicle(taxi);
	lot->addVehicle(truck);
	cout << "Parking Lot Details:" << endl;
	lot->printVehicles();
	cout << "Horn each vehicle in the parking lot:" << endl;
	lot->hornEach();
	delete car;
	delete taxi;
	delete truck;
	delete lot;
	return 0;
}
